docs/2_auth.md

```diff
@@ -85,6 +86,33 @@ Note: The user is checked against the group members list on initial authenticati
 
 Note: When using the Azure Auth provider with nginx and the cookie session store you may find the cookie is too large and doesn't get passed through correctly. Increasing the proxy_buffer_size in nginx or implementing the [redis session storage](configuration#redis-storage) should resolve this.
 
+### Azure OIDC Auth Provider
+
+1. Go to [https://portal.azure.com](https://portal.azure.com), and create new "App Registration".
+1. Go to **"Authentication"** page and configure addresses for your applications: `https://internal.yourcompany.com/oauth2/callback`
+1. Go to **"Manifest"** page and enable group support in ID tokens by adding groupMembershipClaims entry 
+    ```json
+    {
+      "groupMembershipClaims": "SecurityGroup",
+      ...
+    }
+    ```
+1. Go to **"Certificates & secrets"** settings, create new secret and store it in safely. After you close page you won't be able to read it again.
+1. Go to **"Overview"** page to check tenant ID and client ID of "App Registration"
+1. Start oauth2-proxy with given settings
+    ```
+    --azure-permitted-groups=<comma separated list of group object ids>
+    --oidc-issuer-url=https://login.microsoftonline.com/<tenant_id>/v2.0
+    --provider=azure-oidc
+   ```
+
+
+NOTE: This provider doesn't support Azure Active Directory (v1.0) endpoints.
+
+NOTE: If `--azure-permitted-groups` is not specified, all logged users are authorized
```

Comment:
It's not immediately obvious to me what this means, does that mean any user of the tennant or any user of Azure? Could we clarify this one a bit better please